apps/api/src/app/integrations/usecases/get-novu-integration/get-novu-integration.usecase.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@nestjs/common';
+import { IntegrationEntity, MessageRepository, IntegrationRepository, ExecutionDetailsRepository } from '@novu/dal';
+import { GetNovuIntegrationCommand } from './get-novu-integration.command';
+import { ChannelTypeEnum, EmailProviderIdEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
+import { startOfMonth, endOfMonth } from 'date-fns';
+
+@Injectable()
+export class GetNovuIntegration {
+  constructor(
+    private messageRepository: MessageRepository,
+    private integrationRepository: IntegrationRepository,
+    private executionDetailsRepository: ExecutionDetailsRepository
+  ) {}
+
+  async execute(command: GetNovuIntegrationCommand): Promise<IntegrationEntity[]> {
+    const providerId = this.getProviderId(command.channelType);
+
+    const sentMessagesCount = await this.executionDetailsRepository.count({
+      status: ExecutionDetailsStatusEnum.SUCCESS,
+      _organizationId: command.organizationId,
+      channel: command.channelType,
+    });
+
+    if (sentMessagesCount > 0) {
+      return [];
+    }
+
+    const activeIntegrationsCount = await this.integrationRepository.count({
+      _organizationId: command.organizationId,
+      active: true,
+      channel: command.channelType,
+    });
+
+    if (activeIntegrationsCount > 0) {
+      return [];
+    }
+
+    const messagesCount = await this.messageRepository.count({
+      channel: command.channelType,
+      _organizationId: command.organizationId,
+      providerId,
+      createdAt: { $gte: startOfMonth(new Date()), $lte: endOfMonth(new Date()) },
+    });
+
+    if (messagesCount === 200) {
```

Comment:
I saw that you tried to create this use-case to be generic, but I'm wondering if this logic shouldn't be specific just for the `email` type, because the other providers might have different restrictions.

Reply:
My thinking was to make it generic now because I do not see how it could be different for other types. maybe the limit can be different but lets look at it when we add next type.